dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -206,8 +206,9 @@ FramesDecoder::FramesDecoder(const std::string &filename)
 
 
 FramesDecoder::FramesDecoder(const char *memory_file, int memory_file_size, bool build_index,
-                             bool init_codecs, int num_frames)
+                             bool init_codecs, int num_frames, const std::string source_info)
   : av_state_(std::make_unique<AvState>()),
+    filename_(source_info),
```

Comment:
```suggestion
                             bool init_codecs, int num_frames, std::string source_info)
  : av_state_(std::make_unique<AvState>()),
    filename_(std::move(source_info)),
```
...or, as suggested earlier, use `std::string_view`.